### PR TITLE
Publishing to public feeds

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR adds support for publishing packages to an Azure Artifacts public feed. The pipeline is modified to produce three different package versions.

* ```x.y.z``` (no suffix)
* ```x.y.z-run.12345``` (package specific to this run)
* ```x.y.z-preview.abc``` (the preview package)

The idea is that upfront in the build we create our packages up front and carry them through the pipeline and publish them at the right time (unpacking and signing where appropriate). This assumes a pipeline that goes from PR through CI and all the way to publishing.

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -9,8 +9,15 @@ jobs:
...
-      - script: "dotnet pack eng/service.proj -o $(Build.ArtifactStagingDirectory) -warnaserror /p:ServiceDirectory=${{parameters.ServiceDirectory}} /p:PublicSign=false ${{ parameters.VersioningProperties }}"
-        displayName: "Build and Package"
+      - script: "dotnet build eng/service.proj -o $(Build.ArtifactStagingDirectory) -warnaserror /p:ServiceDirectory=${{parameters.ServiceDirectory}} /p:PublicSign=false ${{ parameters.VersioningProperties }}"
+        displayName: "Build"
+        env:
+          DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+          DOTNET_CLI_TELEMETRY_OPTOUT: 1
+          DOTNET_MULTILEVEL_LOOKUP: 0
+      - script: |
+          dotnet pack eng/service.proj --no-build -o $(Build.ArtifactStagingDirectory) -warnaserror /p:ServiceDirectory=${{parameters.ServiceDirectory}} /p:PublicSign=false ${{ parameters.VersioningProperties }} /p:VersionSuffix=run.$(Build.BuildID)"
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml]: Perhaps this is still a work in progress but I only see this producing a single package.

**Comment by weshaggard** [comment]
@mitchdenny lets please chat more about this before you merge anything to enable this.

**Comment by mitchdenny** [comment]
Closing this as it'll be picked up as part of the unified pipelines work.

**Pull request closed without merging**